101-iothub-auto-route-messages/azuredeploy.json

```diff
@@ -0,0 +1,172 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "IoTHubNamePrefix": {
+            "type": "string",
+            "defaultValue": "ContosoTestHub",
+            "maxLength": 37,
+            "metadata": {
+                "description": "The name prefix of the IoT Hub to create."
+            }
+        },
+        "location": {
+            "type": "string",
+            "defaultValue": "[resourceGroup().location]",
+            "metadata": {
+                "description": "The datacenter to use for the deployment."
+            }
+        },
+        "skuName": {
+            "type": "string",
+            "defaultValue": "S1",
+            "metadata": {
+                "description": "The SKU to use for the IoT Hub."
+            }
+        },
+        "skuUnits": {
+            "type": "string",
+            "defaultValue": "1",
+            "metadata": {
+                "description": "The number of IoT Hub units."
+            }
+        },
+        "d2cPartitions": {
+            "type": "string",
+            "defaultValue": "4",
+            "metadata": {
+                "description": "Partitions used for the event stream."
+            }
+        },
+        "storageAccountNamePrefix": {
+            "type": "string",
+            "defaultValue": "demostorage",
+            "maxLength": 11,
+            "metadata": {
+                "description": "Name prefix of storage account to be created."
+            }
+        },
+        "storageContainerName": {
+            "type": "string",
+            "defaultValue": "contosoresults",
+            "metadata": {
+                "description": "Name of the container in which to place the routed data."
```

Comment:
this must be lower case (so you'll have to toLower it) - or you can use a var so the user doesn't enter any disallowed chars